graphql/validate.lua

```diff
@@ -35,6 +46,51 @@ local visitors = {
     rules = { rules.uniqueFragmentNames, exit = { rules.noUnusedFragments } }
   },
 
+  -- <operation>
+  --
+  -- {
+  --   kind = 'operation',
+  --   operation = 'query' or 'mutation',
+  --   name = {
+  --     kind = 'name',
+  --     value = <string>,
+  --   } (optional),
+  --   selectionSet = <selectionSet>,
+  --   variableDefinitions = list of <variableDefinition> (optional),
+  --   directives = list of <directive> (optional),
+  -- }
+  --
+  -- <variableDefinition>
+  --
+  -- {
+  --   kind = 'variableDefinition',
+  --   variable = <variable>,
+  --   type = <type>,
+  --   defaultValue = <value> (optional),
+  -- }
+  --
+  -- <type> is one of the following:
+  --
+  -- * <namedType>
+  -- * <listType>
+  -- * <nonNullType>
+  --
+  -- <namedType>
+  --
+  -- {
+  --   kind = 'namedType',
+  --   name = {
+  --     kind = 'name',
+  --     value = <string>,
+  --   },
+  -- }
+  --
+  -- <listType>, <nonNullType>
+  --
+  -- {
+  --   kind = one of ('listType', 'nonNullType'),
```

Comment:
According to this definition, `nonNullType` may be `nonNullType` (which is recursive), instead it should be one of `namedType` or `listType`

Reply:
It seems correct. Fixed.